docs: Add user guide for CLP MCP server.

rishikeshdevsot opened the pull request:
<!-- markdownlint-disable MD012 -->

<!--
Set the PR title to a meaningful commit message that:

* is in imperative form.
* follows the Conventional Commits specification (https://www.conventionalcommits.org).
  * See https://github.com/commitizen/conventional-commit-types/blob/master/index.json for possible
    types.

Example:

fix: Don't add implicit wildcards ('*') at the beginning and the end of a query (fixes #390).
-->

# Description

<!-- Describe what this request will change/fix and provide any details necessary for reviewers. -->
This request will update user documentation to add:
1. A guide to run the MCP server as part of CLP package
2. Guides to connect Claude Desktop, Cursor and VSCode to CLP MCP server. 



# Checklist

<!-- Ensure each item below is satisfied and indicate so by inserting an `x` within each `[ ]`. -->

* [x] The PR satisfies the [contribution guidelines][yscope-contrib-guidelines].
* [x] This is a breaking change and that has been indicated in the PR title, OR this isn't a
  breaking change.
* [x] Necessary docs have been updated, OR no docs need to be updated.

# Validation performed

<!-- Describe what tests and validation you performed on the change. -->



[yscope-contrib-guidelines]: https://docs.yscope.com/dev-guide/contrib-guides-overview.html


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **Documentation**
  * Added a comprehensive MCP Server guide covering prerequisites, configuration, startup, compressed logs and usage.
  * Added step‑by‑step integration guides for connecting Claude Desktop, Cursor and Visual Studio Code to the MCP server.
  * Included per‑client connection examples, verification steps and sample client configuration snippets.
  * Added the MCP Server guide to the guides index/navigation and external reference links.
  * Noted that log timestamps are treated as UTC and timezone encoding support is planned.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

20001020ycx reviewed (dismissed):
Clear documentation, left a minor comment, the rest LGTM

rishikeshdevsot commented:
> A few high-level comments before a complete review:
> 
> * Can you manually apply the markdown linter (we don't have it automated yet, sorry) to the new files in this PR:
>   ```shell
>   npx markdownlint-cli2 --config tools/yscope-dev-utils/exports/lint-configs/.markdownlint-cli2.yaml --fix <file-path>
>   ```
> * Where we mention external tools like Claude Desktop, can we add links to them? Usually, we link the first instance of each tool on each page.

Fixed linting issues and added the links for all the tools. (the images are not loading on my end for some reason, looking into it now)

rishikeshdevsot commented:
> > Fixed linting issues and added the links for all the tools. (the images are not loading on my end for some reason, looking into it now)
> 
> Do you have Git LFS installed? https://github.com/y-scope/clp/tree/main/docs#requirements

yeah I do, I am working on top of a clp-build-container so all the dependencies are present. I think the images were corrupted locally for some reason, I just re-uploaded them on my local system and it's fine now. 
It shouldn't affect this PR though, I can see the correct images in the `Files` tab of this PR.

rishikeshdevsot commented on docs/src/user-docs/guides-mcp-server/index.md:
```diff
@@ -0,0 +1,91 @@
+    :::{note}
+    Setting `mcp_server: null` or leaving the `mcp_server` section commented will
+    start `clp-json` without the MCP server.
+    :::
```
I don't know if this note is needed... but it was asked for in #1481

20001020ycx commented on docs/src/user-docs/guides-mcp-server/index.md:
````diff
@@ -0,0 +1,90 @@
+   `etc/clp-config.yml` and then specifying a host and port number:
+
+    ```yaml
+    mcp_server:
````
@rishikeshdevsot @kirkrodrigues

kirkrodrigues replied:
We can use a comment instead.

```yaml
mcp_server:
  host: "localhost"
  port: 8080
  # other settings
```

20001020ycx replied:
sounds good to me!

rishikeshdevsot replied:
addressed

kirkrodrigues commented on docs/src/user-docs/guides-mcp-server/index.md:
```diff
@@ -0,0 +1,103 @@
+Agents sometimes ignore MCP instructions even after making the `get_instructions` tool call.
+In your prompt, tell the agent to read the MCP instructions carefully
+before analyzing any logs.
+:::
```
```suggestion
:::{note}
Agents sometimes ignore MCP instructions even after making the `get_instructions` tool call. To
reduce erroneous queries from the agent, in your prompt you can tell the agent to read the MCP
instructions carefully before analyzing any logs.
:::
```

kirkrodrigues commented on docs/src/user-docs/guides-mcp-server/index.md:
```diff
@@ -0,0 +1,103 @@
+CLP currently doesn't store timezone information in the compressed logs. This means that if the
+logs you compressed were *not* in the UTC timezone, you will need to tell your agent what timezone
+the logs were in originally. Support for encoding timezone information in the compressed logs will
+be available in a future release.
```
```suggestion
CLP currently doesn't store timezone information in the compressed logs. This means that if the logs
you compressed were *not* in the UTC timezone, you will need to tell your agent what timezone the
logs were in originally. Support for encoding timezone information in the compressed logs will be
available in a future release.
```

kirkrodrigues commented on docs/src/user-docs/guides-mcp-server/mcp-claude-desktop.md:
```diff
@@ -0,0 +1,59 @@
+# Connecting Claude Desktop
+
+This guide explains how to connect [Claude Desktop][claude-desktop]
+to the CLP MCP server over HTTP.
```
```suggestion
This guide explains how to connect [Claude Desktop][claude-desktop] to the CLP MCP server over HTTP.
```

kirkrodrigues commented on docs/src/user-docs/guides-mcp-server/mcp-claude-desktop.md:
```diff
@@ -0,0 +1,59 @@
+This guide explains how to connect [Claude Desktop][claude-desktop]
+to the CLP MCP server over HTTP.
+
+## Prerequisites
```
```suggestion
## Requirements
```

kirkrodrigues commented on docs/src/user-docs/guides-mcp-server/mcp-claude-desktop.md:
```diff
@@ -0,0 +1,59 @@
+1. Install Node.js >= 16
+2. Install the [mcp-remote] package
+
+## Steps
```
```suggestion
## Setup
```

kirkrodrigues commented on docs/src/user-docs/guides-mcp-server/mcp-claude-desktop.md:
```diff
@@ -0,0 +1,59 @@
+    :::
+
+3. Add the MCP server: Add an entry to the `mcpServers` object. If the `mcpServers`
+object does not exist, create one.
```
```suggestion
3. Add the MCP server to the `mcpServers` object. If the `mcpServers` object does not exist, create
   one.
```

kirkrodrigues commented on docs/src/user-docs/guides-mcp-server/mcp-claude-desktop.md:
```diff
@@ -0,0 +1,59 @@
+    The `host` and `port` values should match the hostname and port number where
+    the MCP server is running, as specified
+    during [configuration](../guides-mcp-server/index.md#starting-the-mcp-server).
+    Check [mcp-remote]'s package description for additional arguments that can be specified.
```
```suggestion
    The `host` and `port` values should match the hostname and port number where the MCP server is
    running, as specified during
    [configuration](../guides-mcp-server/index.md#starting-the-mcp-server). Check the [mcp-remote]'s
    package description for additional arguments that can be specified.
```

kirkrodrigues commented on docs/src/user-docs/guides-mcp-server/mcp-claude-desktop.md:
```diff
@@ -0,0 +1,59 @@
+5. Restart Claude Desktop: Close and reopen the application for the
+new configuration to take effect.
+6. You should now see the MCP server and its available tools under the
+"Search and Tools" icon in Claude Desktop's Prompt window.
```
```suggestion
4. Save changes to the `claude_desktop_config.json` file.
5. Close and reopen Claude Desktop for the new configuration to take effect.
6. You should now see the MCP server and its available tools under the "Search and Tools" icon in
   Claude Desktop's Prompt window.
```

kirkrodrigues commented on docs/src/user-docs/guides-mcp-server/mcp-vscode.md:
```diff
@@ -0,0 +1,34 @@
+
+1. Follow the
+[official VSCode guide][vscode-guide]
+to add an MCP server to VSCode.
```
```suggestion
1. Follow the [official Cursor guide][cursor-guide] to add a Remote MCP Server using `mcp.json`.
```

kirkrodrigues commented on docs/src/user-docs/guides-mcp-server/mcp-vscode.md:
```diff
@@ -0,0 +1,34 @@
+
+    The `host` and `port` values should match the hostname and
+    port number where the MCP server is running, as
+    specified during [configuration](../guides-mcp-server/index.md#starting-the-mcp-server).
```
```suggestion
    The `host` and `port` values should match the hostname and port number where the MCP server is
    running, as specified during
    [configuration](../guides-mcp-server/index.md#starting-the-mcp-server).
```

kirkrodrigues commented on docs/src/user-docs/guides-mcp-server/mcp-cursor.md:
```diff
@@ -0,0 +1,32 @@
+
+    The `host` and `port` values should match the hostname and port number
+    where the MCP server is running, as specified
+    during [configuration](../guides-mcp-server/index.md#starting-the-mcp-server).
```
```suggestion
    The `host` and `port` values should match the hostname and port number where the MCP server is
    running, as specified during
    [configuration](../guides-mcp-server/index.md#starting-the-mcp-server).
```

kirkrodrigues commented on docs/src/user-docs/guides-mcp-server/mcp-cursor.md:
```diff
@@ -0,0 +1,32 @@
+    during [configuration](../guides-mcp-server/index.md#starting-the-mcp-server).
+
+3. Enable and use the MCP server's tools as described in
+the [official Cursor documentation][cursor-using-mcp].
```
```suggestion
3. enable and use the mcp server's tools as described in the [official cursor
   documentation][cursor-using-mcp].
```
